impl/src/try_from.rs

```diff
@@ -0,0 +1,149 @@
+//! Implementation of a [`TryFrom`] derive macro.
+
+use proc_macro2::{Literal, Span, TokenStream};
+use quote::{format_ident, quote, ToTokens as _};
+use syn::{spanned::Spanned as _, Ident, Variant};
+
+/// Expands a [`TryFrom`] derive macro.
+pub fn expand(input: &syn::DeriveInput, _: &'static str) -> syn::Result<TokenStream> {
+    match &input.data {
+        syn::Data::Struct(data) => Err(syn::Error::new(
+            data.struct_token.span(),
+            "`TryFrom` cannot be derived for structs",
+        )),
+        syn::Data::Enum(data) => {
+            let Some(repr) = ReprAttribute::parse_attrs(&input.attrs)? else {
+                return Err(syn::Error::new(
+                    Span::call_site(),
+                    "`TryFrom` needs repr specified `#[repr(u*/i*)]`",
+                ));
+            };
```

Comment:
This is probably overly conservative, we should be able to assume that it is an `isize` if nothing was specified:

> Under the [default representation](https://doc.rust-lang.org/reference/type-layout.html#the-default-representation), the discriminant is interpreted as an isize value. However, the compiler is allowed to use a smaller type (or another means of distinguishing variants) in its actual memory layout.

https://doc.rust-lang.org/reference/items/enumerations.html#discriminants

Reply:
Got another confirmation on the `isize` for the expression especially: https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/enum.20discriminant.20size/near/388599445